catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/IngestCommand.java

```diff
@@ -201,128 +225,81 @@ protected Object executeWithSubject() throws Exception {
+        executorService.submit(new Runnable() {
+            @Override
+            public void run() {
+                buildQueue(ingestStream, metacardQueue, start);
             }
-
-            while (!executorService.isTerminated()) {
-                try {
-                    TimeUnit.MILLISECONDS.sleep(100);
-                } catch (InterruptedException e) {
-                    // ignore
-                }
+        });
+
+        final ScheduledExecutorService batchScheduler =
+                Executors.newSingleThreadScheduledExecutor();
+        executorService.submit(new Runnable() {
+            @Override
+            public void run() {
+                submitToCatalog(batchScheduler,
+                        executorService,
+                        metacardQueue,
+                        catalog,
+                        batchSize,
+                        start);
             }
+        });
 
-            printProgressAndFlush(start, fileCount.get(), ingestCount.get() + ignoreCount.get());
-
-            long end = System.currentTimeMillis();
-
-            console.println();
-            String elapsedTime = timeFormatter.print(new Period(start, end).withMillis(0));
-            console.printf(" %d file(s) ingested in %s %n", ingestCount.get(), elapsedTime);
-
-            LOGGER.info("{} file(s) ingested in {} [{} records/sec]",
-                    ingestCount.get(),
-                    elapsedTime,
-                    calculateRecordsPerSecond(ingestCount.get(), start, end));
-            INGEST_LOGGER.info("{} file(s) ingested in {} [{} records/sec]",
-                    ingestCount.get(),
-                    elapsedTime,
-                    calculateRecordsPerSecond(ingestCount.get(), start, end));
-
-            if (fileCount.get() != ingestCount.get()) {
-                console.println();
-                if ((fileCount.get() - ingestCount.get() - ignoreCount.get()) >= 1) {
-                    String failedAmount = Integer.toString(
-                            fileCount.get() - ingestCount.get() - ignoreCount.get());
-                    printErrorMessage(failedAmount
-                            + " file(s) failed to be ingested.  See the ingest log for more details.");
-                    INGEST_LOGGER.warn("{} files(s) failed to be ingested.", failedAmount);
-                }
-                if (ignoreList != null) {
-                    String ignoredAmount = Integer.toString(ignoreCount.get());
-                    printColor(Ansi.Color.YELLOW,
-                            ignoredAmount
-                                    + " file(s) ignored.  See the ingest log for more details.");
-                    INGEST_LOGGER.warn("{} files(s) were ignored.", ignoredAmount);
-                }
+        while (!doneBuildingQueue.get() || processingThreads.get() != 0) {
+            try {
+                TimeUnit.SECONDS.sleep(2);
+            } catch (InterruptedException e) {
+                // ignore
```

Comment:
Good point! If this exception occurs, the executor services will likely be shutdown prematurely. Added Logging here.